### PR TITLE
Add optional support for Gerrit Trigger Plugin

### PR DESCRIPTION
**Pull request by Jimilian**
After that it would be possible to get quite useful information:
- how much time is needed to get +1 from creating change moment
- how much patch-set do we verify everyday
- every -1/+1 from Gerrit Trigger Plugin
- who creates the most of load
- etc.

### CODE REVIEW

**Comment by Jimilian** [comment]
@heremaps/cci

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
      * @param jsonFromExtension json that would be used as extension
      * @param jsonFromFile json that would be extend
      * @param startTimeInMillis job start time in milliseconds
-     * @throws IllegalArgumentException
+     * @throws IllegalArgumentException if so.
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java]: I don't believe this wording change is a good one as-is. You should explain *what* the "so" is, i.e. the exact condition under which an `IllegalArgumentException` is thrown.

File: src/main/webapp/help-GerritTriggerPluginListener.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    </tr>
+    <tr>
+        <td>event_created_on</td><td>1507275000000</td>
+    </tr>
+    </tbody>
+</table>
+For <code>onStarted</code> (this event is triggered in the moment when Gerrit Trigger Plugin sends information that builds started) it will push the same data except result field.
+<p>In case of not a change related changes (i.e. RefUpdated) only <code>command</code> and <code>result</code> will be sent.
+But it should never happen according the current implementation of GerritTriggerPlugin</p>
+
```
Comment (sschuberth) [inline src/main/webapp/help-GerritTriggerPluginListener.html]: Many superfluous blank lines. Please remove them, but keep a final trailing newline.

File: src/test/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        when(change.getBranch()).thenReturn("branch_name");
+        when(change.getProject()).thenReturn("project_name");
+        when(change.getTopic()).thenReturn("topic_name");
+        when(patchSet.getNumber()).thenReturn("number");
+        when(patchSet.getKind()).thenReturn(GerritChangeKind.TRIVIAL_REBASE);
+        when(patchSet.getCreatedOn()).thenReturn(new Date(10));
+
+        return patchsetCreated;
+    }
+}
```
Comment (sschuberth) [inline src/test/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListenerTest.java]: Missing final trailing newline.

File: src/main/java/org/jenkinsci/plugins/fluentd/LoggerHolder.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package org.jenkinsci.plugins.fluentd;
+
+import org.fluentd.logger.FluentLogger;
+
+import java.util.logging.Logger;
+
+/**
+ * Holds the reference to the current logger
+ */
+final class LoggerHolder {
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/LoggerHolder.java]: As it's not strictly related to the Gerrit functionality, I believe the LoggerHolder related changes should go to a separate preceding commit.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/plugins/fluentd/LoggerHolder.java (hunk 4 above)
Comment (Jimilian) [reply src/main/java/org/jenkinsci/plugins/fluentd/LoggerHolder.java]: In this case it would not be possible to test Gerrit functionality.

**Comment by Jimilian** [comment]
@sschuberth, please, take a look.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**